projects/plugins/jetpack/modules/masterbar/masterbar/class-masterbar.php

```diff
@@ -93,23 +91,39 @@ class Masterbar {
 	 * Constructor
 	 */
 	public function __construct() {
+		$this->user_id = get_current_user_id();
+
+		if ( ! Jetpack::is_user_connected( $this->user_id ) ) {
+			return;
+		}
+
+		$this->user_data       = Jetpack::get_connected_user_data( $this->user_id );
```

Comment:
Any idea about how can we flush the cache used on this function for Jetpack sites? In D56006-code, I updated the cache flushing mechanism that we implemented for the color scheme used on Atomic sites so the cache is also flushed when the link destination setting changes. Unfortunately, that only works on Atomic sites.

@jeherve Is there any way we can trigger a cache flushing of a Jetpack site from WP.com? I guess we can implement a new XMLRPC method that is executed on the Jetpack site and make a XMLRPC request from the WP.com side?

@Automattic/ajax Should we worry about invalidating the cache of Jetpack sites when the opt-out Calypso links setting changes? Or just having covered the Atomic sites is enough?

Reply:
That's a good question. Maybe we should introduce a routine to handle that every time we update the plugin?

@fgiannar @leogermani What would be your take on this?

Reply:
A few clarifications: 
- The link destination setting has not been implemented yet, but it will be a new setting stored on the Calypso preferences (which are stored in a WP.com user attribute) that will be managed in wordpress.com/me/account (as shown [here](https://github.com/Automattic/wp-calypso/issues/47043#issuecomment-723127808)).
- We're exposing the value of the link destination setting in the `wpcom.getUser` XMLRPC response (D56006-code).
- In this PR we're reading the value of the link destination setting, and storing its value on the `jetpack_admin_menu_link_destination` user option, which is used by the Masterbar module (#18238).
- When a user changes the link destination setting in Calypso, the `jetpack_admin_menu_link_destination` user option should immediately change in the request to the  `wpcom.getUser` XMLRPC endpoint.
- Unfortunately, the request to the  `wpcom.getUser` XMLRPC endpoint may not happen until 24 hours later due to the caching mechanism of the Jetpack connection manager package.
- We sorted this out on Atomic sites by running a SSH command that deletes the `jetpack_connected_user_data_<id>` transient every time the link destination setting changes in Calypso (see D56006-code).

Reply:
On a second thought, this may be not something we should be worried about:
- Nav Unification is never enabled on Jetpack sites, so Jetpack users visiting WP Admin will always get the standard WP Admin menu with all menu items linking to `/wp-admin`.
- Jetpack users visiting Calypso will always have a "Dashboard" menu item that links to `/wp-admin/index.php`.

These two ways (which are exclusive to Jetpack sites) offer enough alternatives for Jetpack users to go to from Calypso to WP Admin if they wish.

There will be a third way, which is the one worked on in this PR (and rest of PRs related to https://github.com/Automattic/wp-calypso/issues/47043), and the one that presents the temporary caching issue described above. But again, since the issue is specific to Jetpack users, and those users already have other ways of going to WP Admin, we can simply ignore it for the time being.